sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java

```diff
@@ -82,7 +85,9 @@ private void configureChannelPipelineHandlers() {
             //  By default, keep alive is enabled on http client
             tcpClient = tcpClient.option(ChannelOption.CONNECT_TIMEOUT_MILLIS,
                 (int) configs.getConnectionAcquireTimeout().toMillis());
-
+            tcpClient = tcpClient.doOnConnected(conn -> {
+                conn.addHandlerLast(new WriteTimeoutHandler(this.httpClientConfig.getRequestTimeout().toMillis(), TimeUnit.MILLISECONDS));
```

Comment:
please make sure for both Gateway and DirectHttps  io request timeout from public surface api is correctly plumbed through to `httpHttpConfig.getRequestTimeout()`